File: js/Day2/dive-part-1.js
const fs = require("fs");
const instructions = fs
  .readFileSync("./js/Day2/input.txt")
  .toString()
  .split("\r\n");

const reader = (instruction) => {
  const { groups } =
    /(?<cmd>forward|down|up)\s*(?<number>\+{0,1}\-{0,1}\d+)/g.exec(instruction);
  return { cmd: groups.cmd, number: Number(groups.number) };
};

let horizontal = 0;
let vertical = 0;
for (let i = 0; i < instructions.length; i++) {
  const { cmd, number } = reader(instructions[i]);
  switch (cmd) {
    case "forward":
      horizontal += number;
      break;
    case "down":
      vertical += number;
      break;
    case "up":
      vertical -= number;
      break;
  }
}

console.log({ horizontal, vertical, multiply: horizontal * vertical });

File: rust/Day2/dive_part_1.rs
use std::env;
use std::fs;

struct Submarine;
impl Submarine {
    pub fn read(instruction: &str) {
        // println!("{:?}", instruction.split(" ").collect::<Vec<&str>>());
        if let [cmd, number] = instruction.split(" ").collect::<Vec<&str>>()[..] {
            // println!("{:?} {:?}", number, cmd.to_string());
            println!("{:?}", number);
            match cmd {
                "forward" => println!("forward"),
                "down" => println!("down"),
                "up" => println!("up"),
                _ => println!("wah!")
            };
        }
    }

    // pub fn dive(instructions: Vec<String>) -> i32 {

    // }
}

fn main() {
    let args: Vec<String> = env::args().collect();
    let filename = &args[1];
    println!("In file {}", filename);
    let contents = fs::read_to_string(filename)
        .expect("Something went wrong reading the file");

    let instructions: Vec<&str> = contents.split("\r\n").collect();
    Submarine::read(instructions[0]);
}